### ISSUE TITLE
GIF Preview feature

### ISSUE THREAD

**Issue by arggh** [sindresorhus/Gifski#136]
I think previewing the gif in the same screen where I tweak the conversion settings is a superior user experience compared to going back and forth between the settings-view and the save-view. Especially, since there is already a video player present in the settings-view.

For example, there could be a button <kbd>Render preview</kbd> or a toggle: <kbd>Original</kbd> - <kbd>Preview</kbd>

If you select <kbd>Preview</kbd>, the original video would be replaced with the gif and a progress indicator, that would let me know whenever gifski is busy converting a new preview. Tweaking the settings would start the preview conversion from scratch.

This would make it effortless to tweak the quality and framerate settings.

Another idea to speed up the preview process: how about rendering just the first frame of the gif with the settings applied? That should be plenty fast, and you would get some perspective into what the final result will look like, given the resolution and quality. Maybe even let the user scrub-select frames and render them as needed?

**Comment by sindresorhus** [comment]
I agree, it's a bit cumbersome at the moment if you need to tweak settings.

> or a toggle: Original - Preview

This sounds like a great idea indeed.

> Another idea to speed up the preview process: how about rendering just the first frame of the gif with the settings applied?

Or maybe 5 frames evenly spread out over the video time range. That way you can quickly get a feel of the image quality of the whole video.

> Maybe even let the user scrub-select frames and render them as needed?

I think the optimal rendering steps would be:

1. Render first frame.
2. Render the next 4 frames, evenly spread out.
3. Start rendering the whole thing for real.
4. Also allow scrubbing to render one frame at the scrubbed position.

If anyone wants to work on this, even just step `1` would be a good improvement.

**Comment by kornelski** [comment]
Because gifski tries to encode differences between frames, non-consecutive frames will look a bit different than in the full anim.

How about starting encoding from the start and allowing viewing everything that has been encoded so far?

**Comment by hazeycode** [comment]
Is anyone already working on this? I can take a shot at it if not

**Comment by mmulet** [comment]
Solved by pull request #329
